mercure.rst

```diff
@@ -95,57 +90,70 @@ The preferred way to configure MercureBundle is using
 
 The clients must also bear a `JSON Web Token`_ (JWT)
 to the Mercure Hub to be authorized to publish updates and, sometimes, to subscribe.
 
-This JWT should be stored in the ``MERCURE_JWT_SECRET`` environment variable.
-
-The JWT must be signed with the same secret key as the one used by
-the Hub to verify the JWT (``!ChangeMe!`` in you use the Local Web Server or
-Symfony Docker).
-Its payload must contain at least the following structure to be allowed to
-publish:
+This token must be signed with the same secret key as the one used by the Hub to verify the JWT (``!ChangeMe!`` in you use the Docker integration).
+This secret key must be stored in the ``MERCURE_JWT_SECRET`` environment variable.
+MercureBundle will use it to automatically generate and sign the needed JWTs.
 
-.. code-block:: json
+If you don't want to use the provided environment variables,
+use the following configuration:
 
-    {
-        "mercure": {
-            "publish": []
-        }
-    }
+.. configuration-block::
 
-Because the array is empty, the Symfony app will only be authorized to publish
-public updates (see the authorization_ section for further information).
+    .. code-block:: yaml
 
-.. tip::
+        # config/packages/mercure.yaml
+        mercure:
+            hubs:
+                default:
+                    url: https://mercure-hub.example.com/.well-known/mercure
+                    jwt:
+                        secret: '!ChangeMe!'
 
-    The jwt.io website is a convenient way to create and sign JWTs.
-    Checkout this `example JWT`_, that grants publishing rights for all *topics*
-    (notice the star in the array).
-    Don't forget to set your secret key properly in the bottom of the right panel of the form!
+    .. code-block:: xml
 
-.. caution::
+        <!-- config/packages/mercure.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <config>
+            <hub
+                name="default"
+                url="https://mercure-hub.example.com/.well-known/mercure"
+            >
+                <jwt secret="!ChangeMe!"/>
+            </hub>
+        </config>
 
-    Don't put the secret key in ``MERCURE_JWT_SECRET``, it will not work!
-    This environment variable must contain a JWT, signed with the secret key.
+    .. code-block:: php
 
-    Also, be sure to keep both the secret key and the JWTs... secrets!
+        // config/packages/mercure.php
+        $container->loadFromExtension('mercure', [
+            'hubs' => [
+                'default' => [
+                    'url' => 'https://mercure-hub.example.com/.well-known/mercure',
+                    'jwt' => [
+                        'secret' => '!ChangeMe!',
+                    ],
+                ],
+            ],
+        ]);
 
-If you don't want to use the provided environment variables,
-use the following configuration:
+Alternatively, it's also possible to pass directly the JWT to use:
```

Comment:
should we recommend using secret over jwt value as it allows managing subscribe and publish topics easier from symfony?

Reply:
It depends of the use case, I changed this section.